### PR TITLE
citusdb test setup with Django

### PR DESCRIPTION
**Pull request by snopoke**
This wasn't as bad as I'd thought. It does add 3-5min to each test run due to the 2 additional DBs that Django runs the migrations over.

This is a PR into #23479

### CODE REVIEW

File: docker/localsettings.py [hunk 1]
```diff
@@ -95,32 +95,50 @@
...
-    'TEST': {
-        'NAME': 'postgres',  # use the same DB for tests (without the 'test_' prefix)
-        'SERIALIZE': False,
+# CitusDB setup is done by Django migrations in the testapps.citus_* apps. These
+# migrations are only run during unit tests.
+DATABASES.update({
+    'citus-ucr': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'DISABLE_SERVER_SIDE_CURSORS': True,
+        'NAME': 'commcare_ucr_citus',  # use postgres DB since otherwise we'd need to create the database on the workers
```
[the remark this answers is not in the document]
Comment (millerdev) [reply docker/localsettings.py]: comment should be removed

File: corehq/sql_db/routers.py [hunk 2]
```diff
@@ -25,6 +25,14 @@
 SYNCLOGS_APP = 'phone'
 AAA_APP = 'aaa'
 
+# this logic is necessary for test database setup
+citus_masters = [db for db in settings.DATABASES.values() if db.get('ROLE', None) == 'citus_master']
+assert len(citus_masters) in (0,1), "Multiple CitusDB masters found"
```
Comment (snopoke) [inline corehq/sql_db/routers.py]:
```suggestion
assert len(citus_masters) in (0, 1), "Multiple CitusDB masters found"
```

File: docker/localsettings.py (hunk 1 above)
Comment (snopoke) [inline docker/localsettings.py]:
```suggestion
        'NAME': 'commcare_ucr_citus',
```

**Comment by emord** [comment]
@snopoke How necessary is this setup to be in a django migratin?. With docker we're pulling in from citus db, I assume they have the citus extension enabled. I don't think it would be terrible to require our commcare-cloud to enable the extension on worker nodes either. Would be great if we could use MIGRATE = false to skip migrations on these worekrs

**Comment by snopoke** [comment]
@emord it's not completely necessary but then there are other things that need to worked round in the tests. Having it in the migrations is only for tests and not for live envs where ansible would do the setup.

This comment encapsulates most of the issues: https://github.com/dimagi/commcare-hq/pull/23479#discussion_r261684385

**Pull request merged**